src/node_data.ts

```diff
@@ -72,33 +73,36 @@ declare global {
  * Initializes a NodeData object for a Node.
  */
 function initData(
-    node: Node, nameOrCtor: NameOrCtorDef, key: Key): NodeData {
-  const data = new NodeData(nameOrCtor, key);
+    node: Node, nameOrCtor: NameOrCtorDef, key: Key, text?: string|undefined): NodeData {
+  const data = new NodeData(nameOrCtor, key, text);
   node['__incrementalDOMData'] = data;
   return data;
 }
 
-
 /**
  * Retrieves the NodeData object for a Node, creating it if necessary.
  */
-function getData(node: Node) {
-  importNode(node);
-  return node['__incrementalDOMData']!;
+function getData(node: Node, key?: Key) {
+  // In this case, the node was server-side rendered.
+  if (!node['__incrementalDOMData'] && key !== undefined) {
```

Comment:
I don't think this if statement actually changes anything. There are 4 paths:

| Already imported? | Undefined key? | result |
| --- | --- | --- |
| false | false | imported, `key = attribute OR key` |
| false | true | imported, `key = attribute OR undefined` |
| true | false | not imported |
| true | true | not imported |

But we know that `key === undefined` in the first case, so it's equivalent to `key = attribute OR key`

Reply:
Agreed